python/plugins/processing/algs/qgis/RasterCalculator.py

```diff
@@ -213,3 +225,69 @@ def processBeforeAddingToModeler(self, algorithm, model):
                         values.append(ValueFromOutput(alg.modeler_name, out.name))
 
         algorithm.params[self.LAYERS] = values
+
+    def mappedNameToLayer(self, lyr, expression, layersDict, context):
+        '''Try to identify if a real layer is mapped in the expression with a symbolic name.'''
+
+        nameToMap = lyr.name()
+
+        # get last scope of the expressionContext because should be that related
+        # with mapped variables
+        # The scope name should be "algorithm_inputs"
+        expContextLastScope = context.expressionContext().lastScope()
```

Comment:
This looks fragile - why not go through the whole expressionContext here, instead of just the last scope?

Reply:
I feel it fragile too... I used lastScope because I feel fragile also the way to refer to the scope via a name that is set statically in some other part of code. BTW I agree, I'll use the current state pointing with name.